### ISSUE TITLE
what is a book

### ISSUE THREAD

**Issue by Gerry5100** [openaudible/openaudible#1311]
When I imported my audiobook directory,  Each book had it's own directory.  That directory could contain a hundred or more mp3 files.
These files belonged to a specific book.  OpenAudible treated each file as a book.  So instead of having a couple hundred books, I how have over 20K.  I dont see how this is usable.

**Comment by openaudible** [comment]
Ok. Might be good to move those files out of the OpenAudible books directory and click Refresh Files. That should restore things.

It is not usually possible to know if a directory contains one book with many files, or is many books. 

There are several strategies we use to mark a directory as either being a subdirectory, or an audiobook directory.

To mark a directory as an audiobook directory, there needs to be a .cue, .jpg, or .png file. 
Or you can create a hidden file called .audiobook that will mark the directory as an audibook directory with multiple files.
If you want a directory to be a subdirectory, you can create a .subdirectory file in a directory. All mp3/m4b files will be treated as books, and all subdirectories will be "guessed".. and can be hinted using the .audiobook or .subdirectory files.

Need to get that into the documentation, and also planning to add that to the GUI so it is easier to figure out.

If all your books are in their own folders, I'm not sure we have a great solution yet to mass importing them. 

Open to ideas or discussion.

**Comment by Gerry5100** [comment]
My current architecture is E:\Audiobooks\Authors\BookTitle\mp3 or E:\Audiobooks\Authors\Series\Booktitle\mp3.  I'm thinking of putting a dummy .jpg file in each booktitle directory where the mp3s are located.  And put a ".subdirectory" file in Aufhors and Series directories.  When I import into OA, I'm hoping there is going to be no copying or duplication going to take place -- I will test this on a very limited scale.

I would suggest offering the user the ability to import books by their directory tree. In other words, make a pointer in the library that points to the books actual location for mp3 or m4a or whatever (audio files).  The files in OA book directory could just be pointers to the actual media content files(mp3).   I was thinking of a Author file .i.e.(Stephen King) this file would contained a list of his book titles, or Series Titles in your library,  The Series Titles files would contained book titles in the series. Roughly, a form of linked list via file names.  I envision that the user could implement a book group just by placing the book titles into something like a book group file. Similar to the Author File, Series File they only point to another file in the library list. Only the bookfile would contain the location of audio files.  This eliminates a need for file movement. Plus, this would prevent a books multiple mp3 files from each being considered as a book.  In order to solve your current mp3/book issue(IMHO) you could create these files in a top directory called catalog. 

Basically,  Author contains AuthorBooks or Series, Series points to

**Comment by Gerry5100** [comment]
Tested the import directory function with the ".subdirectory" placed in each directory that was a subdirectory.  Directory Tree  Home/Mediafiles/AuthorAA/Ann_Rand
                            .subdirectory 
                                            .sudirectory
                                                             /book1
                                                                jpgfile
                                                                AAXfile
                                                             /book2
                                                                jpgfile
                                                                AAXfile
                                             John_Jones
                                               .subdirectory
                                                                  jpgfile
                                                                  book3
                                             SeriesZZ
                                              .subdirectory
                                                                Books11/
                                                                             jpgfile
                                                                             mp3A
                                                                             mp3B
                                                                Books22/
                                                                              jpgfile
                                                                              mp3C
                                                                              mp3D

This structure is copied into OA books directory and then you are informed of the media files it found. The program produces and
error screen stating that it could not create mediafiles because it already existed. I think it is trying to create mediafiles file that would contain the books.  This behaviour is consistent with any directory that contains ".subdirectory".
